lib/stages/VisSharedMemWriter.cpp

```diff
@@ -72,6 +69,28 @@ VisSharedMemWriter::~VisSharedMemWriter() {
     }
 }
 
+bool VisSharedMemWriter::wait_for_semaphore() {
+    // handles timed waits for semaphores
+    // does a standard wait if the system clock is not accessible
+
+    struct timespec ts;
+    if (clock_gettime(CLOCK_REALTIME, &ts) == -1){
+        DEBUG("Failed to get system time.\n Not using timed semaphores.");
```

Comment:
Could print `std::strerror(errno)` here to find out what's going on: https://linux.die.net/man/3/clock_gettime